allow lines before json header

maks opened the pull request:
This  allows the json header in files to start on lines other than the first one, allowing for bracketing the json with Jeykll style front-matter `---` marker lines which means editors like VS Code will not show the header when doing a preview of a markdown file, eg.
```markdown
---
{
  "layout": "post",
  "title": "My new post",
  "tags": ["programming"],
  "date": "Jan 16 2019",
  "img" : "logo.png",
  "alt" : "logo image"
}
---
# My new post

```

![Screenshot from 2020-01-16 15-37-50](https://user-images.githubusercontent.com/71999/72493829-2ec23f80-3876-11ea-9505-5adc01d53bb6.png)


This PR also updates the package versions in the lock file to latest matching from npm.

mantoni commented:
Thank you for contributing! Your change causes two tests to fail, please check. Also, would you care to add an additional test to demonstrate your use case?

maks commented:
@mantoni thanks for looking at it so quickly and apologies for rushing without checking the tests.
I've changed the implementation to handle the front-matter markers a bit better and added a test for it now too.

mantoni commented:
Released as `v1.1.0`.
